### ISSUE TITLE
Prometheus exporter handles delta metrics differently from the current spec

### ISSUE THREAD

**Issue by dashpole** [open-telemetry/opentelemetry-java#5015]
**Describe the bug**

The prometheus exporter [appears to](https://github.com/open-telemetry/opentelemetry-java/blob/f490ba8ee6f36fb7a3395c41f6b7673b279a97d7/exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusType.java#L35) convert sums with aggregation temporality delta to gauges, and [appears to ignore](https://github.com/open-telemetry/opentelemetry-java/blob/f490ba8ee6f36fb7a3395c41f6b7673b279a97d7/exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Serializer.java#L186) temporality for histograms.

**What did you expect to see?**

The [spec for Sums says](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md#sums):

> OpenTelemetry Sums follows this logic:
> * If the aggregation temporality is cumulative and the sum is monotonic, it MUST be converted to a Prometheus Counter.
> * If the aggregation temporality is cumulative and the sum is non-monotonic, it MUST be converted to a Prometheus Gauge.
> * If the aggregation temporality is delta and the sum is monotonic, it SHOULD be converted to a cumulative temporality and become a Prometheus Sum. The following behaviors are expected:
>   * The new data point type must be the same as the accumulated data point type.
>   * The new data point's start time must match the time of the accumulated data point. If not, see detecting alignment issues.
>   * Otherwise, it MUST be dropped.

The [spec for Histograms says](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md#histograms-1):

> OpenTelemetry Histograms with Delta aggregation temporality SHOULD be aggregated into a Cumulative aggregation temporality and follow the logic above, or MUST be dropped.

Basically, we need to either convert delta -> cumulative, or drop the metrics.

It is also definitely possible that i'm mistaken and it isn't possible for delta metrics to be passed to the Prometheus exporter.  In that case, we should probably remove the [aggregation temporality check](https://github.com/open-telemetry/opentelemetry-java/blob/f490ba8ee6f36fb7a3395c41f6b7673b279a97d7/exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusType.java#L35) from the metric -> type logic.

**Comment by jack-berg** [comment]
> It is also definitely possible that i'm mistaken and it isn't possible for delta metrics to be passed to the Prometheus exporter.

This. Cumulative temporality for all instruments is specified [here](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServer.java#L117-L119). It will be possible for `MetricProducer`s other than the SDK to ignore the temporality, but the Prometheus reader [won't be required](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricreader) to do anything about it:

> The MetricReader is not required to ensure data points from a non-SDK [MetricProducer](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricproducer) are output in the configured aggregation temporality, as these data points are not collected using OpenTelemetry instruments.

**Comment by dashpole** [comment]
Thanks.  Based on the prometheus spec above, I would still expect delta temporality metrics to be dropped in the exporter.  At minimum, I wouldn't expect delta sums to be converted to gauges.  But this is moot until after https://github.com/open-telemetry/opentelemetry-java/issues/5011
